test/function.localstack.test.ts

```diff
@@ -712,7 +779,54 @@ const testFunction = async (
     })
     .promise();
 
-  expect(
-    result.Payload ? JSON.parse(result.Payload.toString()) : undefined
-  ).toEqual(expected);
+  try {
+    expect(
+      result.Payload ? JSON.parse(result.Payload.toString()) : undefined
+    ).toEqual(expected);
+  } catch (e) {
+    console.error(result);
+    throw e;
+  }
 };
+
+test("should not create new resources in lambda", async () => {
+  await expect(
+    async () => {
+      const stack = new Stack();
+      new Function(
+        stack,
+        "function",
+        {
+          timeout: Duration.seconds(20),
+        },
+        async () => {
+          const bus = new aws_events.EventBus(stack, "busbus");
+          return bus.eventBusArn;
+        }
+      );
+      await Promise.all(Function.promises);
+    }
+    // TODO: add error message
```

Comment:
Should we catch this in the language service too?

Reply:
What would we do? Warn when app.synth() is used?

Reply:
Was just thinking of an error code in the IDE

Reply:
This is for creating constructs in the function right?

Reply:
Ohh, yeah. Yeah, we can detect new on Construct or Resource (we should create a FunctionlessResource) interface or something.

Reply:
This test was just moved in this PR. https://github.com/functionless/functionless/issues/246